### PR TITLE
Final Clean Up of 2.0 Series P2MM/Preparing For 2.2 P2MM Release

### PR DESCRIPTION
**Pull request by OrsellGaming**
## Introduction

The current P2MM `main` and `dev` branches are a mess and Valve has forced our hand at releasing a new version. These repositories need cleanup before we release the pre-release/beta that will allow people to play P2MM again. This PR primarily aims to fix parts of this repositorie’s unstable mess and make it more usable for those who want to fork it. Also, this is here to add a few other QoL things and general updates to everything with new features and bug fixes. This PR will be merged into `dev`, then `dev` will merge into `main`, so both branches will be cleaned.

For those interested in contributing to the final 2.2 version of the 2.0 series of P2MM, we highly recommend you use the `finalcleanup` branch for development. `dev` will still exist after this PR is merged for any other bug fixes or new translations are created after 2.2 releases. It would be great also you could also help us with the Goals/Tasks we have listed below. We will not accept new features or any significant changes after 2.2 releases unless they're really needed.

We also need some help testing all these new changes, so if you want to help us with testing, feel free to fork the repository and run the P2MM Launcher and mod on the `finalcleanup` branch. If you don't want to use the source code, you can contact `zwexit\ (Orsell)` in our Discord server, and he'll give you a precompiled executable depending on your system. If you do get this, though, please help test and report any bugs, as well as any suggestions. If you find an issue or bug, or have a suggestion, please post an issue post in the Issues tab of this repository.

## Goals/Tasks:

- [x] General code cleanup.
- [x] Add more translations.
- [x] Better support for the Gelocity maps.
- [x] Fix common mounting bugs.
- [x] Update Launcher Icon.
- [x] Remove the fallback system, as it is not needed anymore.
- [x] Update ModFiles.
- [x] Fix the clipboard support on Linux.
- [x] Fix Issue #45 .
- [x] Implement Issue #77 .
- [x] Add music control in `mp_coop_lobby_3`.
- [x] Modify the chat commands behavior to command behavior like in Minecraft. `!command (target) (operation/destination/argument)` or if no `target` `!command (operation/destination)`
- [ ] An investigation has shown that the Linux plugin is the source of the problems causing Linux users to crash as Valve has made updates to Portal 2, making our byte patches incorrect to what is there. This is currently being fixed by @0xNULLderef 
- [ ] Finish Steam Deck support. Get a custom control configuration setup for Steam Deck's Gaming Mode. @OrsellGaming 
- [x] Create an AppImage script for Linux. Being worked on by @cabiste69 
- [ ] Test different Linux distros to ensure they are working correctly.
- [x] Figure out how to get all the Valve credits to show up after the P2MM credits. The P2MM credits cause the Valve credits to not line up correctly, so the last chunk of Valve credits doesn't get displayed.
- [x] Update the launcher update system to download the right AppImage architecture for Linux. We also have the problem of the `2.1.0` launcher being unable to download the new AppImages because it's not programmed to look for them. @cabiste69 said he would help with this.
- [x] Check if the warning for xclip/wl-clipboard is missing appears correctly in Italian. @WhatDidYouExpect 
- [x] Fix the crash when a button does not have attribute X. @cabiste69 
- [x] Fix README header HTML to fix all the shield positions and properly size the logo in the header.
- [x] Have the launcher run frame updates to the launcher while in an InputPrompt or when a toast message appears at the bottom. @cabiste69 will do this because I don't want to accidentally screw up something.
- [ ] REMEMBER TO UPDATE THE MODINDEX!!!

THIS PULL REQUEST IS BEING MERGED EARLY DUR TO VALVE MAKING AN UPDATE FOR PORTAL 2 BREAKING OUR MOD!!! `2.2.0` will be pre-released then gradually updated until a full release. Everything not done here had been moved onto a todo list else where. Work will resume in `dev` branch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,13 +98,14 @@ We are always happy to take whatever additional help we can get for this project
 - vista
 - Wolƒe Strider Shoσter
 - cabiste
-- zwexit (Orsell) 
 
 **Contributors:**
+- zwexit\ (Orsell)
```
Comment (cabiste69) [inline README.md:103]: bro really put his name above wanderer 😭

Comment (OrsellGaming) [reply README.md:103]: My man, all i did was cut and paste it under contributor, it ain't that deep.

**Comment by cabiste69** [comment]
why did you remove the files under `src/DevFiles/Tools/`? and especially the script that updates the identifier file?
you also renamed the identifier file yet forgot to rename it in the updater module

**Comment by OrsellGaming** [comment]
Yeah, I realized that just after I committed. I'm currently working on getting it back. My initial intention of removing the dev files was because we didn't need them anymore or they were old and obsolete, essentially trying to reduce the file size of source code by removing any unnessisary files. I'm gonna readd the files that are needed.

**Comment by OrsellGaming** [comment]
It was a simple accidental whoopsy do.

**Comment by OrsellGaming** [comment]
# Update about this pull request!
So I'm close to getting everything done, I've just got a few things left to do, but I'm going on a family trip soon, so I will have to put some stuff on hold and wrap them up later. The list above of things I want to do in the original PR message is obsolete as I've got new info to share, so the list here will contain the last things I need to do.

- [ ] Wrap up Steam Deck support: Despite what my commit title has said, I'm still working on it again as I ran into several new issues. YIPPIE!!! So Desktop Mode and Gaming Mode behave differently with the P2MM Launcher because why would they function the same. So using the on-screen keyboard in Gaming Mode, the shift button and caps lock work, but the arrow buttons, which I had initially used for shifting and pasting, don't work. I could do myself a favor and remove pasting entirely, which would knock out a few other issues, but then using the P2MM Launcher in Desktop Mode without a keyboard and mouse will be a pain. There is also the problem where in Gaming Mode, I have to have a controller layout for it to work. I still need to figure out if I can share it publically for people to use.
- [ ] Appimage for Linux systems: I still intend to make an Appimage for Linux systems which will allow it to have an icon and be more compatible and allow not to need to chmod the file to execute. Although I still believe you must click "Execute" on the dialogue that appears. Before, if I recall correctly, it just did nothing because it didn't have the proper permissions.
- [ ] Update the ModFiles: I still need to update the ModFiles, although I've been met with many different options for going through with it. 1. Copy and paste in the dev branch's newer but less stable ModFiles; while less stable, it also has better-optimized code by Nano and other updated and new aspects that have been handy. Again though, there could be stability problems and crashes. 2. Update the map support files; the dev branch's map support files have been updated and improved, so doing those could help. Although I might have to make some changes to all of them because the plugin created some functions that don't exist using the old plugin. 3. Leave the ModFiles as they are; while I least like this option because it doesn't have new content, it's the most stable option. If I choose this option, I need to fix the map support file for sp_a4_intro at least because it is currently broken, causing all P2MM functionality to break and spawn users in a box outside the map.

##
So these are all the changes I want to wrap up. P2MM is then ready for a 2.2.0 release—a final major update to the old launcher. The only possibility of another release would be a bug fix update or if someone wants to add a translation to the launcher. However, I could ChatGPT a few more in as I had to do that for updating the other translations. We'll see.

**Comment by 0xNULLderef** [comment]
uhhh sorry didn't mean to edit lmao

**Comment by OrsellGaming** [comment]
Opened as this is going to be merged with `dev`, then `dev` will merge with `main`, then both branches will be cleaned up for the `2.2.0-beta.1` pre-release beta.

File: src/Scripts/BasicFunctions.py [hunk 2]
```diff
@@ -1,81 +1,212 @@
...
-            print(manifestpath)
-            if (os.path.isfile(manifestpath)):
-                # read the manifest file
-                f = open(manifestpath, "r", encoding="utf-8")
-                manifest = f.read()
-                f.close()
+            steamPath = winreg.QueryValueEx(hkey, "InstallPath")
+            print(steamPath)
+            manifestPath = steamPath[0] + NormalizePath("/steamapps/libraryfolders.vdf")
+            print(manifestPath)
```
Comment (cabiste69) [inline src/Scripts/BasicFunctions.py:201]: these prints should be removed

Comment (OrsellGaming) [reply src/Scripts/BasicFunctions.py:201]: you added those, ill dev-mode them

File: src/ModFiles/Portal 2/install_dlc/scripts/vscripts/multiplayermod/config.nut [hunk 3]
```diff
@@ -18,23 +12,30 @@
...
-// 0 - Standard   : A stable experience
-// 1 - Speedrun   : For those who like to speedrun Portal 2     (No implementation yet)
-// 2 - Deathmatch : Fight to the death in a team battle         (No full implementation yet)
-// 3 - Futbol     : Portal 2 beta minigame restored             (No full implementation yet)
+// 0 - Standard   : Most stable experience
+// 1 - Speedrun   : For those who like to speedrun Portal 2     (No full implementation yet)
 Config_GameMode <- 0
 
-// Set true/false if you want to randomize every Turret's models and colors (for fun lol)
+// Set true/false if you want to randomize every Turret's models and colors (for fun lol) (BROKEN D:)
```
Comment (cabiste69) [inline src/ModFiles/Portal 2/install_dlc/scripts/vscripts/multiplayermod/config.nut:38]: 💔

Comment (OrsellGaming) [reply src/ModFiles/Portal 2/install_dlc/scripts/vscripts/multiplayermod/config.nut:38]: I think I could probably make it work again, I'll add it to my to do list.

Comment (OrsellGaming) [reply src/ModFiles/Portal 2/install_dlc/scripts/vscripts/multiplayermod/config.nut:38]: Yes, we are getting close

**Review by cabiste69 — APPROVED** [review]
looks good

**Comment by OrsellGaming** [comment]
Ok then, gonna merge. Anything else?

**Comment by OrsellGaming** [comment]
It is time to merge `finalcleanup` into `dev`! Then `dev` into `main`.

**Pull request merged**